timeseries-streaming/timeseries-python-applications/MLPipeline/timeseries/encoder_decoder/encoder_decoder_run_fn.py

```diff
@@ -29,18 +30,19 @@ def _input_fn(
         tf_transform_output: tft.TFTransformOutput,
         batch_size: int = 2) -> tf.data.Dataset:
     transformed_feature_spec = (
-            tf_transform_output.transformed_feature_spec().copy())
+        tf_transform_output.transformed_feature_spec().copy())
 
     dataset = tf.data.experimental.make_batched_features_dataset(
-            file_pattern=file_pattern,
-            batch_size=batch_size,
-            # TODO Document. Shuffle must be set to False.
-            shuffle=False,
-            features=transformed_feature_spec,
-            reader=_gzip_reader_fn)
+        file_pattern=file_pattern,
+        batch_size=batch_size,
+        # Shuffle must be False, as Zip operation is not transitive
+        shuffle=False,
+        features=transformed_feature_spec,
+        reader=_gzip_reader_fn)
```

Comment:
you can use [tfxio](https://github.com/tensorflow/tfx/blob/master/tfx/examples/iris/iris_utils_native_keras.py#L89) to get dataset now

Reply:
Which is the preferred option? Any pros / cons?

Reply:
tfxio is perferred, we changed all major examples to use tfxio now

Reply:
@1025KB will switch over to tfxio in a separate commit, after this branch is merged, which we will target for this week.